test.py

```diff
@@ -1,87 +1,122 @@
-from sys import argv
-from pathlib import Path
-from os import walk, system
-from ast import literal_eval
 import subprocess
+from ast import literal_eval
+from dataclasses import dataclass
+from enum import Enum
+from os import system
+from pathlib import Path
+from sys import argv
+
+
+class ExpectedOutputType(Enum):
+    """Possible test results"""
+
+    FAIL = 1
+    EXIT_WITH_CODE = 2
+    EXIT_WITH_OUTPUT = 3
+
+
+@dataclass(frozen=True)
+class Expected:
+    """A container for the expected output of a test"""
+
+    expected_output_type: ExpectedOutputType
```

Comment:
Maybe rename this field to `type`? Seems unnecessarily verbose.

Reply:
ce6be68